rpc/internal/buildlogger_service.go

```diff
@@ -68,8 +69,36 @@ func (s *buildloggerService) AppendLogLines(ctx context.Context, lines *LogLines
 		newRPCError(codes.Internal, errors.Wrapf(log.Append(exportedLines), "problem appending log lines for '%s'", lines.LogId))
 }
 
-func (s *buildloggerService) StreamLog(stream Buildlogger_StreamLogServer) error {
-	return nil
+// StreamLogLines adds log lines via client-side streaming to an existing
+// buildlogger log.
+func (s *buildloggerService) StreamLogLines(stream Buildlogger_StreamLogLinesServer) error {
+	ctx := stream.Context()
+	id := ""
+
+	for {
+		if err := ctx.Err(); err != nil {
+			return newRPCError(codes.Aborted, err)
+		}
+
+		lines, err := stream.Recv()
+		if err == io.EOF {
+			return stream.SendAndClose(&BuildloggerResponse{LogId: id})
+		}
+		if err != nil {
+			return err
+		}
+
+		if id == "" {
+			id = lines.LogId
+		} else if lines.LogId != id {
+			return newRPCError(codes.Aborted, errors.New("log ID in stream does not match reference, aborting"))
+		}
+
+		_, err = s.AppendLogLines(ctx, lines)
+		if err != nil {
+			return err
```

Comment:
should we put an RPC error code on this?

Reply:
it calls AppendLogLines which sets the rpc codes, so no need to :)